tests/d3d12.c

```diff
@@ -33843,6 +33843,535 @@ static void test_constant_buffer_sm51(void)
 }
 
+static void test_bindless_buffer_srv_sm51(void)
+{
+    D3D12_ROOT_SIGNATURE_DESC root_signature_desc;
+    D3D12_ROOT_PARAMETER root_parameters[3];
+    D3D12_DESCRIPTOR_RANGE descriptor_ranges[2];
+    ID3D12DescriptorHeap* heap;
+
+    ID3D12Resource *input_buffers[1024];
+    ID3D12Resource *output_buffer;
+    struct resource_readback rb;
+
+    ID3D12GraphicsCommandList* command_list;
+    D3D12_CPU_DESCRIPTOR_HANDLE cpu_handle;
+    D3D12_GPU_DESCRIPTOR_HANDLE gpu_handle;
+    unsigned int i, descriptor_size;
+    struct test_context context;
+    ID3D12CommandQueue* queue;
+    HRESULT hr;
+
+    if (!init_compute_test_context(&context))
+        return;
+    command_list = context.list;
+    queue = context.queue;
+
+    root_signature_desc.NumParameters = 3;
+    root_signature_desc.Flags = 0;
+    root_signature_desc.NumStaticSamplers = 0;
+    root_signature_desc.pStaticSamplers = NULL;
+    root_signature_desc.pParameters = root_parameters;
+
+    root_parameters[0].ParameterType = D3D12_ROOT_PARAMETER_TYPE_DESCRIPTOR_TABLE;
+    root_parameters[0].ShaderVisibility = D3D12_SHADER_VISIBILITY_ALL;
+    root_parameters[0].DescriptorTable.NumDescriptorRanges = 1;
+    root_parameters[0].DescriptorTable.pDescriptorRanges = &descriptor_ranges[0];
+
+    root_parameters[1].ParameterType = D3D12_ROOT_PARAMETER_TYPE_DESCRIPTOR_TABLE;
+    root_parameters[1].ShaderVisibility = D3D12_SHADER_VISIBILITY_ALL;
+    root_parameters[1].DescriptorTable.NumDescriptorRanges = 1;
+    root_parameters[1].DescriptorTable.pDescriptorRanges = &descriptor_ranges[1];
+
+    root_parameters[2].ParameterType = D3D12_ROOT_PARAMETER_TYPE_UAV;
+    root_parameters[2].ShaderVisibility = D3D12_SHADER_VISIBILITY_ALL;
+    root_parameters[2].Descriptor.RegisterSpace = 0;
+    root_parameters[2].Descriptor.ShaderRegister = 0;
+
+    descriptor_ranges[0].RegisterSpace = 1;
+    descriptor_ranges[0].BaseShaderRegister = 4;
+    descriptor_ranges[0].OffsetInDescriptorsFromTableStart = 0;
+    descriptor_ranges[0].NumDescriptors = 64 * 1024;
+    descriptor_ranges[0].RangeType = D3D12_DESCRIPTOR_RANGE_TYPE_SRV;
+
+    descriptor_ranges[1].RegisterSpace = 2;
+    descriptor_ranges[1].BaseShaderRegister = 4;
+    descriptor_ranges[1].OffsetInDescriptorsFromTableStart = 1;
+    descriptor_ranges[1].NumDescriptors = 64 * 1024;
+    descriptor_ranges[1].RangeType = D3D12_DESCRIPTOR_RANGE_TYPE_SRV;
+
+    hr = create_root_signature(context.device, &root_signature_desc, &context.root_signature);
+    ok(SUCCEEDED(hr), "Failed to create root signature, hr %#x.\n", hr);
+
+    static const DWORD cs_code[] = {
+#if 0
+        StructuredBuffer<uint> Buffers[] : register(t4, space1);
+        StructuredBuffer<uint> AliasBuffers[64] : register(t4, space2);
+        StructuredBuffer<uint> StandaloneBuffer1 : register(t100, space2);
+        StructuredBuffer<uint> StandaloneBuffer2 : register(t200, space2);
+
+        RWByteAddressBuffer OBuffer : register(u0);
```

Comment:
It might be useful to throw in a texture array for completeness. Doesn't have to be anything complex, but maybe something like
```
Texture2D<float4> Textures[2] : register(t0, space3);
// ...
result += uint(Textures[1].Load(int2(0,0)).x);
```
where the first texture is black and the second one is white or something.

Reply:
Added tests for this.